stdlib/Test/src/logical_tree.jl

```diff
@@ -0,0 +1,341 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+"""
+    LogicalTreeTests
+
+For showing logical test failures informatively.
+"""
+module LogicalTreeTests
```

Comment:
This is more code than the actual change and its tests combined - is this really worth adding like that?

Reply:
You mean don't add it as a new module?

Reply:
The primary motivation to do this was because it broke a test in Compiler/SSAIR, while Test tests were fine. Knowing how much nesting stuff in the ecossystem needs feels hard to draw a line for. And for the Compiler test failure, fixing the previous approach proved quite difficult to get right even at that level of nesting, once you factor in negation on nested logic. This just felt like the correct way to go, though I agree it's a bit of a sledgehammer. Maybe this kind of thing is helpful elsewhere?

Reply:
I also thought this could make it easier to add logical text styling.